crates/rolldown_plugin_dynamic_import_vars/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+          Option::<TSTypeParameterInstantiation>::None,
+          false,
+        );
+        self.current += 1;
+      }
+    }
+  }
+}
+
+impl<'ast, 'a> DynamicImportVarsVisit<'ast, 'a> {
+  /// generates helper function declaration
+  fn helper_func(
+    &self,
+    name: &String,
+    files: std::vec::Vec<String>,
+    import_arg: Option<&Expression<'ast>>,
+  ) -> Declaration<'ast> {
+    self.ast_builder.declaration_function(
+      FunctionType::FunctionDeclaration,
+      SPAN,
+      Some(self.ast_builder.binding_identifier(SPAN, name)),
+      false,
+      false,
+      false,
+      Option::<TSTypeParameterDeclaration>::None,
+      None,
+      self.ast_builder.formal_parameters(
+        SPAN,
+        FormalParameterKind::FormalParameter,
+        self.ast_builder.vec1(self.ast_builder.formal_parameter(
+          SPAN,
+          self.ast_builder.vec(),
+          self.ast_builder.binding_pattern(
+            self.ast_builder.binding_pattern_kind_binding_identifier(SPAN, "path"),
+            Option::<TSTypeAnnotation>::None,
+            false,
+          ),
+          None,
+          false,
+          false,
+        )),
+        Option::<BindingRestElement>::None,
+      ),
+      Option::<TSTypeAnnotation>::None,
+      Some(self.ast_builder.function_body(
+        SPAN,
+        self.ast_builder.vec(),
+        self.ast_builder.vec1(self.ast_builder.statement_switch(
+          SPAN,
+          self.ast_builder.expression_identifier_reference(SPAN, "path"),
+          self.switch_cases(files, import_arg),
+        )),
+      )),
+    )
+  }
+
+  /// generates:
+  /// ```js
+  /// case "./file.js": return import("./file.js");
+  /// ```
+  fn switch_cases(
+    &self,
+    files: std::vec::Vec<String>,
+    import_arg: Option<&Expression<'ast>>,
+  ) -> Vec<'ast, SwitchCase<'ast>> {
+    let mut items = self.ast_builder.vec_with_capacity(files.len() + 1);
+    files.into_iter().for_each(|file| {
+      items.push(self.ast_builder.switch_case(
+        SPAN,
+        Some(self.ast_builder.expression_string_literal(SPAN, file.clone())),
+        self.ast_builder.vec1(self.ast_builder.statement_return(
+          SPAN,
+          Some(self.ast_builder.expression_import(
+            SPAN,
+            self.ast_builder.expression_string_literal(SPAN, file),
+            import_arg.map_or_else(
+              || self.ast_builder.vec(),
+              |arg: &Expression<'ast>| self.ast_builder.vec1(clone_expr(self.ast_builder, arg)),
+            ),
+          )),
+        )),
+      ));
+    });
+    items.push(self.default_case());
+    items
+  }
+
+  /// generates:
+  /// ```js
+  /// default: return new Promise(function(resolve, reject) {
+  ///   (typeof queueMicrotask === 'function' ? queueMicrotask : setTimeout)(
+  ///     reject.bind(null, new Error("Unknown variable dynamic import: " + path))
+  ///   );
+  /// })
+  /// ```
+  fn default_case(&self) -> SwitchCase<'ast> {
```

Comment:
The pr implement is good for me, but we need to align `vite-plugin-dynamic-import-vars`
 plugin behavior, the `@rollup/plugin-dynamic-import-vars
` is not suitable for vite, see reason at https://github.com/vitejs/vite/pull/7756.

The `vite-plugin-dynamic-import-vars` plugin will use  an `dynamicImportHelper` module to handle fallback behavior, it will avoid we re-create default case for every situation. And it will transform glob pattern to `import.meta.glob()` to handle files match, it will avoid same logic for create `switch_cases` for files.

Could you adjust some behavior to vite. Thank you.